### PR TITLE
Update macOS clang for Xcode 10

### PR DESCRIPTION
**Pull request by saagarjha**
This pull request:

1. Opens files as UTF-8 so we don't fail on files with Unicode in them
2. Adds newer Apple clangs, with version numbers based on `_LIBCPP_VERSION`.

### CODE REVIEW

File: plugin/tools.py [hunk 1]
```diff
@@ -37,7 +37,8 @@
     '8.1': '3.9',
     '8.2': '3.9',
     '9.0': '4.0',
-    '9.1': '4.0'
+    '9.1': '5.0',
```
Comment (niosus) [inline plugin/tools.py]: Why is this changed? Are you sure this is the correct way?

File: plugin/tools.py [hunk 2]
```diff
@@ -712,7 +713,7 @@ def get_clang_version_str(cls, clang_binary):
     @classmethod
     def _get_regular_clang_version_str(cls, output_text):
         # now we have the output, and can extract version from it
-        version_regex = re.compile(r"\d\.\d\.*\d*")
+        version_regex = re.compile(r"\d*\.\d*\.\d*")
```
Comment (niosus) [inline plugin/tools.py]: Can we use + and not * here? I feel it will be more safe.

File: plugin/tools.py [hunk 3]
```diff
@@ -723,12 +724,12 @@ def _get_regular_clang_version_str(cls, output_text):
 
     @classmethod
     def _get_apple_clang_version_str(cls, output_text):
-        version_regex = re.compile(r"\d\.\d\.*\d*")
+        version_regex = re.compile(r"\d*\.\d*\.\d*")
```
Comment (niosus) [inline plugin/tools.py]: Same as above.

Comment (darkfader) [reply plugin/tools.py]: yeah, + is better than *.
Also needs something like `osx_version = '.'.join(version_str.split('.')[:2])` to get first two numbers.

**Comment by niosus** [comment]
For the lack of activity here I will close this in favor of #483

**Comment by saagarjha** [comment]
Sorry about that; for a while I was unfortunately not able to make code contributions here for a while. I'm glad that you were able to merge in the changes nonetheless. In regards to modifying the version of LLVM used by Xcode 9.1, I just copied whatever `_LIBCPP_VERSION` gave me. So I'm not sure that this is quite correct, but it seems to be more accurate than just guessing that the underlying version would be.
